Task instantiation

veerajsk opened the pull request.

veerajsk committed:
Compute dynamic entrypoints before task assignment

abhi1287 commented on alica_engine/src/engine/model/Plan.cpp:
```diff
@@ -106,9 +105,34 @@ void Plan::setTransitions(const TransitionGrp& transitions)
+    }
+    constexpr std::size_t MAX_CACHE_SIZE = 1000;
+    while (_entryPointCache.size() > MAX_CACHE_SIZE) {
+        _entryPointCache.erase(_entryPointCache.begin());
```
This can remove valid recent entries + doesn't scale well. Once cache reaches 1000 entries, this will likely hit every time there are new dynamic id's.  On reaching max cache size, you can iterate over latest dynamic id's  and remove everything from cache that's not valid. + take max_cache_size from alica.yaml

abhi1287 replied:
+ entryPoint is stored as const pointer, you will have to delete it first or better store as unique_ptr

veerajsk replied:
- yes, I have put a TODO to decide the cache policy
- Will store as unique_ptr

veerajsk replied:
implemented lru caching

veerajsk committed:
Use lru caching for entry point cache

veerajsk committed:
Include dynamic ids in the plan tree info msg

veerajsk committed:
Use context from root to leaf for success semantics

Success semantics should include the entire context from the root
to the leaf of the running plan instead of just the entry point
information. The context of a running plan is the hash of <dynamic
entry point id, state id, parent's hash>. If a running plan is
successful the context of success is the hash of <parent running plan's
context, entry point id, dynamic entry point id>. This commint includes
the changes to only set the success. Reading the success information
from other agents & subsequent changes to the success data structures
is not included.

veerajsk committed:
Include succeeded contexts in the plan tree info msg

veerajsk committed:
Adapt code to new plan tree info msg

veerajsk committed:
Use 0 for parent hash for master (root)

veerajsk committed:
Conside parent context when checking successes in plan

veerajsk committed:
Use the parent's context to remove success marks

abhi1287 commented:
@veerajsk  please mark the pr ready for review when you wan it reviewed

veerajsk commented:
> @veerajsk please mark the pr ready for review when you wan it reviewed

sure will do

veerajsk committed:
Store the SuccessCollection per context

christianhelm committed:
Implement hash functions (#193)

* Implement hash functions

* Add comment

veerajsk committed:
Supply parent context hash for assignment

veerajsk committed:
Merge remote-tracking branch 'origin/feature/task_instantiation' into feature/task_instantiation

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
Fix compile errors

veerajsk committed:
Temp

christianhelm committed:
Context hash fix (#195)

* Implement hash functions

* Add comment

* Change hash function

* Fix hash_combine

veerajsk committed:
Merge remote-tracking branch 'origin/feature/task_instantiation' into feature/task_instantiation

veerajsk committed:
Implement entry point store

veerajsk committed:
Temp

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
First compiling version

veerajsk committed:
Fixes

bjoernschroeder committed:
Add getApplicationEntrypointContext method to basicPlans (#194)

* add getApplicationEntrypointContext to BasicPlan

* change method signature, add default implementation

* return bool, receive entryPointMap in getApplicationEntrypointContext

david-simoes-93 committed:
Dynamic Task Instantiation Unit Testing (#204)

* Crashfixes

* Added a new Plan for UnitTesting the DynamicTask Assignment

* Added test fixture

* Code generation

* Fixed test

* Updated ALICA UI version

* Code generation

* IsDynamic flag in PLM

* CodeGeneration

* Integrating getApplicationEntrypointContext

veerajsk committed:
Get dynamic eps from autogen code invocation

veerajsk committed:
Bug fixes

veerajsk committed:
Bug fixes

veerajsk committed:
Fix dynamic flag name

veerajsk committed:
Fixes

veerajsk committed:
WIP

veerajsk committed:
Fixes

veerajsk committed:
Fix cardinality

veerajsk committed:
Format

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
Fixes

veerajsk committed:
don't update assignment if dynamic id does not match

david-simoes-93 committed:
PlanTree Serialization (#205)

* implement serialization and deserialization

* start implementation for serialization unit test

* Test fixed

* Compared PlanTrees in DynamicEPs

* Creating dynEP in TestPlan and updating test

* fix increment, check for size of ids before getting the currentStateId

* move comment

* implement serialization tests

* format

Co-authored-by: bjoernschroeder <bschroederprogramming@gmail.com>

Commit:
Add plans to test simple plan tree generation

veerajsk committed:
Gen code

veerajsk committed:
fixes

veerajsk committed:
add deserialization tc

veerajsk committed:
Add unit test with multiple tasks in same plan

bjoernschroeder committed:
pass parentContext to updateSuccessCollection (#223)

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
fix serialization tests

veerajsk committed:
Add unit test for test assignment

bjoernschroeder committed:
change reexport plans, set masterPlan flags (#226)

veerajsk committed:
Fix context hash calc, add success unit test

veerajsk committed:
Merge remote-tracking branch 'origin/feature/task_instantiation' into feature/task_instantiation

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
Implement clearing of success marks

veerajsk committed:
Merge branch 'rr_devel' into feature/task_instantiation

veerajsk committed:
Fix getting basic plan from plan pool

veerajsk committed:
Fix initialization

bjoernschroeder committed:
Fix assignment test (#227)

* fix assignment test

* wait max 5 seconds for plan to be active

* assert that agent is in desired plan after waiting

* lock setting running to false

bjoernschroeder committed:
Success marks tests (#233)

* add test for clearing success marks

* add test for receiving successMarks of other agents

* add tests for own successMarks and other agent successMarks

* implement transitions to finish state

* finish implementation of unittests

* add additional check

* format

bjoernschroeder committed:
TaskInstantiation integration test (#255)

* add summand for assigning payloads

* add worldmodel

* generate and implement plans and behaviours

* add pml files

* add integration test, change CMakeLists.txt

* add configs for two more agents

* Use one dynEp per payload, calculate distance between payload and agent

* add assignPayload and freePayload behaviour, split navigation behaviour into drop and pick

* check for dynamicEpId in getAgentsWorking

* step agents 10 times max

* use signed int for dynamicEpId

* fix AssignPayloadSummand, implement FreePayload behaviour, finish integration test implementation, clean up plans and behaviours

* clean code

* format

* rename PickDrop to MovePayload, rename FreePayload to UnassignPayload, clean up Pick behaviour

* remove unused variable

* move default value to header

* use payloadId, iterate through payloads

* add shared worldModel

* remove assignPayload and unassignPayload behaviours, use shared worldmodel

* fix dynId, remove unused variable

* format
